### ISSUE TITLE
Update to v1.2.35 failing

### ISSUE THREAD

**Issue by Roshanalexander** [valiquette/homebridge-orbit-irrigation#94]
Updating from v1.2.33 to v1.2.35 is failing. Error log:

```
USER: administrator
DIR: /var/lib/homebridge
CMD: /opt/homebridge/lib/node_modules/homebridge-config-ui-x/plugin-upgrade-install.sh homebridge-orbit-irrigation 1.2.35 /var/lib/homebridge/node_modules
Target Version: 1.2.35
Target Path: /var/lib/homebridge/node_modules
Downloading homebridge-orbit-irrigation-1.2.35.sha256...
Downloading homebridge-orbit-irrigation-1.2.35.tar.gz...
###=-
###=-

######                           22.5%

###
############################ 100.0%

Verifying download...
homebridge-orbit-irrigation-1.2.35.tar.gz: OK

Creating backup...
Backup path: /var/lib/homebridge/node_modules/.homebridge-orbit-irrigation-1.2.35.bak

Extracting...
./
./homebridge-orbit-irrigation/
./homebridge-orbit-irrigation/.gitattributes
./homebridge-orbit-irrigation/index.js
./homebridge-orbit-irrigation/orbitupdate.js
./homebridge-orbit-irrigation/orbitapi.js
./homebridge-orbit-irrigation/config.schema.json
./homebridge-orbit-irrigation/devices/
./homebridge-orbit-irrigation/devices/battery.js
./homebridge-orbit-irrigation/devices/valve.js
./homebridge-orbit-irrigation/devices/sensor.js
./homebridge-orbit-irrigation/devices/switch.js
./homebridge-orbit-irrigation/devices/irrigation.js
./homebridge-orbit-irrigation/devices/bridge.js
./homebridge-orbit-irrigation/LICENSE
./homebridge-orbit-irrigation/README.md
./homebridge-orbit-irrigation/CHANGELOG.md
./homebridge-orbit-irrigation/package.json
./homebridge-orbit-irrigation/logo/
./homebridge-orbit-irrigation/logo/bhyve.png
./homebridge-orbit-irrigation/logo/homebridge-bhyve.png
./homebridge-orbit-irrigation/orbitplatform.js
./homebridge-orbit-irrigation/node_modules/
./homebridge-orbit-irrigation/node_modules/follow-redirects/
./homebridge-orbit-irrigation/node_modules/follow-redirects/debug.js
./homebridge-orbit-irrigation/node_modules/follow-redirects/http.js
./homebridge-orbit-irrigation/node_modules/follow-redirects/index.js
./homebridge-orbit-irrigation/node_modules/follow-redirects/https.js
./homebridge-orbit-irrigation/node_modules/follow-redirects/LICENSE
./homebridge-orbit-irrigation/node_modules/follow-redirects/README.md
./homebridge-orbit-irrigation/node_modules/follow-redirects/package.json
./homebridge-orbit-irrigation/node_modules/combined-stream/
./homebridge-orbit-irrigation/node_modules/combined-stream/yarn.lock
./homebridge-orbit-irrigation/node_modules/combined-stream/lib/
./homebridge-orbit-irrigation/node_modules/combined-stream/lib/combined_stream.js
./homebridge-orbit-irrigation/node_modules/combined-stream/Readme.md
./homebridge-orbit-irrigation/node_modules/combined-stream/License
./homebridge-orbit-irrigation/node_modules/combined-stream/package.json
./homebridge-orbit-irrigation/node_modules/proxy-from-env/
./homebridge-orbit-irrigation/node_modules/proxy-from-env/index.js
./homebridge-orbit-irrigation/node_modules/proxy-from-env/LICENSE
./homebridge-orbit-irrigation/node_modules/proxy-from-env/README.md
./homebridge-orbit-irrigation/node_modules/proxy-from-env/.travis.yml
./homebridge-orbit-irrigation/node_modules/proxy-from-env/.eslintrc
./homebridge-orbit-irrigation/node_modules/proxy-from-env/package.json
./homebridge-orbit-irrigation/node_modules/proxy-from-env/test.js
./homebridge-orbit-irrigation/node_modules/form-data/
./homebridge-orbit-irrigation/node_modules/form-data/lib/
./homebridge-orbit-irrigation/node_modules/form-data/lib/populate.js
./homebridge-orbit-irrigation/node_modules/form-data/lib/form_data.js
./homebridge-orbit-irrigation/node_modules/form-data/lib/browser.js
./homebridge-orbit-irrigation/node_modules/form-data/Readme.md
./homebridge-orbit-irrigation/node_modules/form-data/README.md.bak
./homebridge-orbit-irrigation/node_modules/form-data/License
./homebridge-orbit-irrigation/node_modules/form-data/package.json
./homebridge-orbit-irrigation/node_modules/form-data/index.d.ts
./homebridge-orbit-irrigation/node_modules/asynckit/
./homebridge-orbit-irrigation/node_modules/asynckit/stream.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/
./homebridge-orbit-irrigation/node_modules/asynckit/lib/state.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/readable_serial_ordered.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/defer.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/readable_serial.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/abort.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/terminator.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/async.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/readable_asynckit.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/streamify.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/iterate.js
./homebridge-orbit-irrigation/node_modules/asynckit/lib/readable_parallel.js
./homebridge-orbit-irrigation/node_modules/asynckit/index.js
./homebridge-orbit-irrigation/node_modules/asynckit/bench.js
./homebridge-orbit-irrigation/node_modules/asynckit/LICENSE
./homebridge-orbit-irrigation/node_modules/asynckit/README.md
./homebridge-orbit-irrigation/node_modules/asynckit/serial.js
./homebridge-orbit-irrigation/node_modules/asynckit/serialOrdered.js
./homebridge-orbit-irrigation/node_modules/asynckit/package.json
./homebridge-orbit-irrigation/node_modules/asynckit/parallel.js
./homebridge-orbit-irrigation/node_modules/axios/
./homebridge-orbit-irrigation/node_modules/axios/dist/
./homebridge-orbit-irrigation/node_modules/axios/dist/node/
./homebridge-orbit-irrigation/node_modules/axios/dist/node/axios.cjs
./homebridge-orbit-irrigation/node_modules/axios/dist/node/axios.cjs.map
./homebridge-orbit-irrigation/node_modules/axios/dist/axios.js.map
./homebridge-orbit-irrigation/node_modules/axios/dist/browser/
./homebridge-orbit-irrigation/node_modules/axios/dist/browser/axios.cjs
./homebridge-orbit-irrigation/node_modules/axios/dist/browser/axios.cjs.map
./homebridge-orbit-irrigation/node_modules/axios/dist/axios.min.js
./homebridge-orbit-irrigation/node_modules/axios/dist/esm/
./homebridge-orbit-irrigation/node_modules/axios/dist/esm/axios.js.map
./homebridge-orbit-irrigation/node_modules/axios/dist/esm/axios.min.js
./homebridge-orbit-irrigation/node_modules/axios/dist/esm/axios.js
./homebridge-orbit-irrigation/node_modules/axios/dist/esm/axios.min.js.map
./homebridge-orbit-irrigation/node_modules/axios/dist/axios.js
./homebridge-orbit-irrigation/node_modules/axios/dist/axios.min.js.map
./homebridge-orbit-irrigation/node_modules/axios/lib/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/common/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/common/utils.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/index.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/node/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/node/index.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/node/classes/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/node/classes/FormData.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/node/classes/URLSearchParams.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/index.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/classes/
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/classes/FormData.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/classes/URLSearchParams.js
./homebridge-orbit-irrigation/node_modules/axios/lib/platform/browser/classes/Blob.js
./homebridge-orbit-irrigation/node_modules/axios/lib/env/
./homebridge-orbit-irrigation/node_modules/axios/lib/env/README.md
./homebridge-orbit-irrigation/node_modules/axios/lib/env/data.js
./homebridge-orbit-irrigation/node_modules/axios/lib/env/classes/
./homebridge-orbit-irrigation/node_modules/axios/lib/env/classes/FormData.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/
./homebridge-orbit-irrigation/node_modules/axios/lib/core/transformData.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/Axios.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/InterceptorManager.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/AxiosHeaders.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/dispatchRequest.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/mergeConfig.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/README.md
./homebridge-orbit-irrigation/node_modules/axios/lib/core/buildFullPath.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/settle.js
./homebridge-orbit-irrigation/node_modules/axios/lib/core/AxiosError.js
./homebridge-orbit-irrigation/node_modules/axios/lib/utils.js
./homebridge-orbit-irrigation/node_modules/axios/lib/defaults/
./homebridge-orbit-irrigation/node_modules/axios/lib/defaults/transitional.js
./homebridge-orbit-irrigation/node_modules/axios/lib/defaults/index.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/deprecatedMethod.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/formDataToJSON.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/isURLSameOrigin.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/speedometer.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/readBlob.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/bind.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/isAxiosError.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/trackStream.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/spread.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/composeSignals.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/buildURL.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/isAbsoluteURL.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/toURLEncodedForm.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/AxiosTransformStream.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/fromDataURI.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/validator.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/cookies.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/parseHeaders.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/README.md
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/ZlibHeaderTransformStream.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/throttle.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/HttpStatusCode.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/callbackify.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/progressEventReducer.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/resolveConfig.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/AxiosURLSearchParams.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/combineURLs.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/toFormData.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/null.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/formDataToStream.js
./homebridge-orbit-irrigation/node_modules/axios/lib/helpers/parseProtocol.js
./homebridge-orbit-irrigation/node_modules/axios/lib/cancel/
./homebridge-orbit-irrigation/node_modules/axios/lib/cancel/CanceledError.js
./homebridge-orbit-irrigation/node_modules/axios/lib/cancel/isCancel.js
./homebridge-orbit-irrigation/node_modules/axios/lib/cancel/CancelToken.js
./homebridge-orbit-irrigation/node_modules/axios/lib/axios.js
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/http.js
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/adapters.js
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/README.md
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/fetch.js
./homebridge-orbit-irrigation/node_modules/axios/lib/adapters/xhr.js
./homebridge-orbit-irrigation/node_modules/axios/index.js
./homebridge-orbit-irrigation/node_modules/axios/LICENSE
./homebridge-orbit-irrigation/node_modules/axios/README.md
./homebridge-orbit-irrigation/node_modules/axios/MIGRATION_GUIDE.md
./homebridge-orbit-irrigation/node_modules/axios/CHANGELOG.md
./homebridge-orbit-irrigation/node_modules/axios/package.json
./homebridge-orbit-irrigation/node_modules/axios/SECURITY.md
./homebridge-orbit-irrigation/node_modules/axios/index.d.ts
./homebridge-orbit-irrigation/node_modules/axios/index.d.cts
./homebridge-orbit-irrigation/node_modules/ws/
./homebridge-orbit-irrigation/node_modules/ws/lib/
./homebridge-orbit-irrigation/node_modules/ws/lib/stream.js
./homebridge-orbit-irrigation/node_modules/ws/lib/websocket.js
./homebridge-orbit-irrigation/node_modules/ws/lib/permessage-deflate.js
./homebridge-orbit-irrigation/node_modules/ws/lib/event-target.js
./homebridge-orbit-irrigation/node_modules/ws/lib/constants.js
./homebridge-orbit-irrigation/node_modules/ws/lib/receiver.js
./homebridge-orbit-irrigation/node_modules/ws/lib/buffer-util.js
./homebridge-orbit-irrigation/node_modules/ws/lib/sender.js
./homebridge-orbit-irrigation/node_modules/ws/lib/websocket-server.js
./homebridge-orbit-irrigation/node_modules/ws/lib/limiter.js
./homebridge-orbit-irrigation/node_modules/ws/lib/subprotocol.js
./homebridge-orbit-irrigation/node_modules/ws/lib/validation.js
./homebridge-orbit-irrigation/node_modules/ws/lib/extension.js
./homebridge-orbit-irrigation/node_modules/ws/index.js
./homebridge-orbit-irrigation/node_modules/ws/wrapper.mjs
./homebridge-orbit-irrigation/node_modules/ws/LICENSE
./homebridge-orbit-irrigation/node_modules/ws/README.md
./homebridge-orbit-irrigation/node_modules/ws/package.json
./homebridge-orbit-irrigation/node_modules/ws/browser.js
./homebridge-orbit-irrigation/node_modules/mime-types/
./homebridge-orbit-irrigation/node_modules/mime-types/index.js
./homebridge-orbit-irrigation/node_modules/mime-types/LICENSE
./homebridge-orbit-irrigation/node_modules/mime-types/README.md
./homebridge-orbit-irrigation/node_modules/mime-types/package.json
./homebridge-orbit-irrigation/node_modules/mime-types/HISTORY.md
./homebridge-orbit-irrigation/node_modules/delayed-stream/
./homebridge-orbit-irrigation/node_modules/delayed-stream/lib/
./homebridge-orbit-irrigation/node_modules/delayed-stream/lib/delayed_stream.js
./homebridge-orbit-irrigation/node_modules/delayed-stream/.npmignore
./homebridge-orbit-irrigation/node_modules/delayed-stream/Readme.md
./homebridge-orbit-irrigation/node_modules/delayed-stream/License
./homebridge-orbit-irrigation/node_modules/delayed-stream/package.json
./homebridge-orbit-irrigation/node_modules/delayed-stream/Makefile
./homebridge-orbit-irrigation/node_modules/mime-db/
./homebridge-orbit-irrigation/node_modules/mime-db/index.js
./homebridge-orbit-irrigation/node_modules/mime-db/db.json
./homebridge-orbit-irrigation/node_modules/mime-db/LICENSE
./homebridge-orbit-irrigation/node_modules/mime-db/README.md
./homebridge-orbit-irrigation/node_modules/mime-db/package.json
./homebridge-orbit-irrigation/node_modules/mime-db/HISTORY.md
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket-cjs.js
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket.mjs
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/events.d.ts
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket-iife.js
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket.d.ts
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket-iife.min.js
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket-amd.js
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/dist/reconnecting-websocket-mjs.js
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/reconnecting-websocket.ts
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/.prettierignore
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/LICENSE
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/README.md
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/CHANGELOG.md
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/package.json
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/CONTRIBUTING.md
./homebridge-orbit-irrigation/node_modules/reconnecting-websocket/events.ts
Extracted to: /var/lib/homebridge/node_modules

Running post-install scripts...
/opt/homebridge/lib/node_modules/npm/lib/cli/validate-engines.js:31
    throw err
    ^

Error: Cannot find module 'graceful-fs'
Require stack:
- /opt/homebridge/lib/node_modules/npm/lib/cli/entry.js
- /opt/homebridge/lib/node_modules/npm/lib/cli.js
- /opt/homebridge/lib/node_modules/npm/bin/npm-cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli/entry.js:15:3)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli/validate-engines.js:39:10)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli.js:4:31)
    at Object.<anonymous> (/opt/homebridge/lib/node_modules/npm/bin/npm-cli.js:2:25)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/homebridge/lib/node_modules/npm/lib/cli/entry.js',
    '/opt/homebridge/lib/node_modules/npm/lib/cli.js',
    '/opt/homebridge/lib/node_modules/npm/bin/npm-cli.js'
  ]
}

Node.js v20.17.0
Failed to rebuild.
Restoring previous version...
Restore Complete. Installation failed.

Bundled install / update could not complete. Trying regular install / update using npm.
USER: administrator
DIR: /var/lib/homebridge
CMD: npm install --save homebridge-orbit-irrigation@1.2.35
/opt/homebridge/lib/node_modules/npm/lib/cli/validate-engines.js:31
    throw err
    ^

Error: Cannot find module 'graceful-fs'
Require stack:
- /opt/homebridge/lib/node_modules/npm/lib/cli/entry.js
- /opt/homebridge/lib/node_modules/npm/lib/cli.js
- /opt/homebridge/lib/node_modules/npm/bin/npm-cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at require (node:internal/modules/helpers:179:18)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli/entry.js:15:3)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli/validate-engines.js:39:10)
    at module.exports (/opt/homebridge/lib/node_modules/npm/lib/cli.js:4:31)
    at Object.<anonymous> (/opt/homebridge/lib/node_modules/npm/bin/npm-cli.js:2:25)
    at Module._compile (node:internal/modules/cjs/loader:1469:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1548:10) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/homebridge/lib/node_modules/npm/lib/cli/entry.js',
    '/opt/homebridge/lib/node_modules/npm/lib/cli.js',
    '/opt/homebridge/lib/node_modules/npm/bin/npm-cli.js'
  ]
}

Node.js v20.17.0


Operation failed with code 7.

You can download this log file for future reference.

See https://github.com/homebridge/homebridge-config-ui-x/wiki/Troubleshooting for help.

```

**Comment by valiquette** [comment]
This update may have trigged this but this looks like a node issue, the error Error: Cannot find module 'graceful-fs' This is not a dependency for this plugin nor any of the modules used in this plug in.

**Comment by Roshanalexander** [comment]
Fixed by downgrading NodeJS to v20.16.0 and then installing the update.
